analyses/doublet-detection/scripts/01a_run-scdblfinder.R

```diff
@@ -0,0 +1,120 @@
+
+
+# Load libraries and renv environment ------
+project_root <- here::here()
+renv::load(project_root)
+
+library(SingleCellExperiment)
+library(optparse)
+
+
+# Functions -----------
+
+#' Run scDblFinder on an SCE object
+#'
+#' @param sce SCE object to run scDblFinder on
+#' @param cores Number of cores to specify to BiocParallel::MulticoreParam(
+#' @param sample_var Required for SCE object with multiple samples, the SCE colData variable that indicates sample of origin
+#' @param random_seed Required for SCE object with multiple samples, a random seed to ensure reproducibility when running on multiple samples
+#' @param ... Additional arguments to pass to scDblFinder::scDblFinder
+#'
+#' @return Data frame object with scDblFinder inferences
+run_scdblfinder <- function(sce,
+                            cores = 4,
+                            sample_var = NULL,
+                            random_seed = NULL,
+                            ...) {
+
+# first check that random seed is set for multiple samples 
+if (is.null(sample_var) && is.null(random_seed)) {
+    # See section 1.5.6: https://bioconductor.org/packages/3.19/bioc/vignettes/scDblFinder/inst/doc/scDblFinder.html#usage
+    stop("If multiple samples are present in the input SCE object, a random seed must be provided for reproducibility.")
+}
+
+# set up cores
+if (cores == 1) {
+  bp <- BiocParallel::SerialParam(RNGseed = random_seed)
+} else {
+  bp <- BiocParallel::MulticoreParam(cores, RNGseed = random_seed)
+}
+
+# Run doublet finder 
+result_df <- scDblFinder::scDblFinder(
+  sce,
+  samples = sample_var # Default is NULL so use whatever was provided by the user or default 
+  BPPARAM = bp,
+  returnType = "table", # return df, not sce
+  ...
+)
+
+  result_df <- result_df |>
+    as.data.frame() |>
+    tibble::rownames_to_column("barcodes")
+
+  return(result_df)
+}
+
+
+# Parse options --------
+
+option_list <- list(
+  make_option(
+    "--dataset_name",
+    type = "character",
+    help = "Name of dataset to process, where the associated file is expected to be named `{name}_sce.rds`"
+  ),
+  make_option(
+    "--data_dir",
+    type = "character",
+    default = "",
+    help = "The directory containing the input RDS file."
+  ),
+  make_option(
+    "--results_dir",
+    type = "character",
+    default = "",
+    help = "The directory to export TSV file with doublet inferences."
+  ),
+  make_option(
+    "--cores",
+    type = "integer",
+    default = 4,
+    help = "Number of cores to use during scDblFinder inference."
+  ),
+  make_option(
+    "--random_seed",
+    type = "integer",
+    default = 2024,
+    help = "Random seed. Default is 2024."
+  )
+)
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# Check input arguments
+if (opts$dataset_name == "") {
+  stop("Must provide a dataset name with --dataset_name.")
+}
+if (!dir.exists(opts$data_dir)) {
+  stop("Must provide an input path to data files with --data_dir.")
+}
```

Comment:
You probably don't need to check both that the data directory exists and the input file exists. I think just the input file is fine, but if you want to leave this also okay.